Update README for pgindent and remove outdated files

amilkh opened the pull request:
Done as a result of a discussion on gpdb-dev:
"Formatting code with CLion in PostgreSQL style"

Signed-off-by: Ben Christel <bchristel@pivotal.io> @benchristel

amilkh commented:
Does anybody have suggestions of what to add under "Updating typedefs.list"?

hlinnaka commented on src/tools/pgindent/README:
```diff
@@ -1,79 +1,64 @@
-	src/tools/find_typedef /usr/local/pgsql/bin /usr/local/pgsql/lib > /tmp/pgtypedefs
+pgindent'ing Greenplum Code
+============================
+Currently, pgindent is run against GPDB before major releases (as does Postgres).
```
Not really. There hasn't been any consistent effort to run pgindent before releases.

amilkh replied:
Updated!

hlinnaka commented on src/tools/pgindent/README:
```diff
@@ -1,79 +1,64 @@
+pgindent, e.g. code originating from Postgres 8.3 should be pgindent'ed using
+an 8.3 pgindent binary so that there are fewer merge conflicts when commits from
+Postgres 8.4 are cherry-picked on top of it.
+For example, `src/backend/main/main.c` is formatted with pgindent 8.3 rules
```
This example is going to go out of date very quickly. Not sure what to give as an example instead that would stay current for longer. Maybe just remove the sentence.

amilkh replied:
Thanks removed.

hlinnaka commented on src/tools/pgindent/README:
```diff
@@ -1,79 +1,64 @@
+GPDB, all code should be kept pgident'ed using the corresponding version of
+pgindent, e.g. code originating from Postgres 8.3 should be pgindent'ed using
+an 8.3 pgindent binary so that there are fewer merge conflicts when commits from
+Postgres 8.4 are cherry-picked on top of it.
```
I would advice to not try running pgindent on any upstream code. It's fiddly to get right. I'd recommend running `git diff` against the upstream version instead, and fixing any spurious differences manually.

amilkh replied:
Hmm, I think if we had a file that originated from Postgres 8.3, but that was modified by a GPDB developer at a later time, it should be `pgindent`ed using the 8.3 binary. `git diff` wouldn't help here, unless that developer changed formatting on the pure upstream portions of the code~

amilkh replied:
Have reworded that section.

hlinnaka commented on src/tools/pgindent/README:
```diff
@@ -1,79 +1,64 @@
+originates from.
+2. Download that version of the Postgres code.
+3. Run pgindent by following the `src/tools/pgindent/README` of the downloaded
+Postgres code, but use the `src/tools/pgindent/typedefs.list` provided with GPDB.
```
See above comment, I would not recommend people to do that. It's too fiddly and cumbersome. You can run `git diff' on a single file against the upstream version, and fix issues manually.

For GPDB-specific code that lives in a file that's inherited from PostgreSQL, I'm not sure what to suggest. Perhaps run latest version pgindent, but only `git add` and commit the changes to the GPDB-specific code. Or just don't bother with pgindent, and try to make the code look nice and more or less the same as what pgindent would produce, manually.

amilkh replied:
> See above comment, I would not recommend people to do that. It's too fiddly and cumbersome. You can run `git diff' on a single file against the upstream version, and fix issues manually.
As above comment, not sure if `git diff` will help here.

>For GPDB-specific code that lives in a file that's inherited from PostgreSQL, I'm not sure what to suggest. Perhaps run latest version pgindent, but only git add and commit the changes to the GPDB-specific code. Or just don't bother with pgindent, and try to make the code look nice and more or less the same as what pgindent would produce, manually.
Hmm, I would think that that file should be `pgindent`ed using the `pgindent` binary version corresponding to the version of Postgres that file originated from. Not a clear answer though, as it could be Frankenstein code...

amilkh commented:
Sorry for the delay in addressing your comments @hlinnaka.
I've updated the README.

@ashwinstar I have added a section regarding `typedefs.list` based on your input from earlier as well.

If no other comments, I will merge this next week.
